coresimd/src/x86/i586/sse2.rs

```diff
@@ -29,7 +28,7 @@ pub unsafe fn _mm_pause() {
 #[inline(always)]
 #[target_feature = "+sse2"]
 #[cfg_attr(test, assert_instr(clflush))]
-pub unsafe fn _mm_clflush(p: *mut c_void) {
+pub unsafe fn _mm_clflush(p: *mut u8) {
```

Comment:
Intel specifies that the [`_mm_clflush` intrinsic](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_clflush&expand=4101,657) takes a `const void*`. I don't know why it takes a `*mut c_void`, and am slightly uncomfortable with it taking a `*mut u8` instead. 

The tracking issue to fix this is: https://github.com/rust-lang/rust/issues/36193

I've opened this issue to track the intrinsics in which we break the API because of the dependency on `c_void`. This should block the stabilization of these intrinsics.

https://github.com/rust-lang-nursery/stdsimd/issues/199